src/AWS.Deploy.Orchestration/Orchestrator.cs

```diff
@@ -131,16 +134,22 @@ public async Task DeployRecommendation(CloudApplication cloudApplication, Recomm
             _interactiveService.LogMessageLine(string.Empty);
             _interactiveService.LogMessageLine($"Initiating deployment: {recommendation.Name}");
 
-            if (recommendation.Recipe.DeploymentType == DeploymentTypes.CdkProject)
-            {
-                _interactiveService.LogMessageLine("AWS CDK is being configured.");
-                await _cdkManager.EnsureCompatibleCDKExists(Constants.CDK.DeployToolWorkspaceDirectoryRoot, Constants.CDK.MinimumCDKVersion);
-            }
-
             switch (recommendation.Recipe.DeploymentType)
             {
                 case DeploymentTypes.CdkProject:
-                    await _cdkProjectHandler.CreateCdkDeployment(_session, cloudApplication, recommendation);
+                    var cdkProject = await _cdkProjectHandler.ConfigureCdkProject(_session, cloudApplication, recommendation);
+                    _interactiveService.LogMessageLine("AWS CDK is being configured.");
+
+                    if (_cdkVersionDetector == null)
+                    {
+                        throw new InvalidOperationException($"{nameof(_cdkVersionDetector)} must not be null.");
+                    }
+
+                    var csprojFiles = Directory.GetFiles(cdkProject, "*.csproj") ?? new string[]{};
```

Comment:
Can we move this into a utility method to get a project file for a directory and then also look for fsproj files? I know it is highly unlikely that somebody will write a CDK project in F# but I want us to in the practice for always looking for either type of project file.

Reply:
100%

Reply:
Fixed